react/utils/pages.ts

```diff
@@ -325,18 +325,29 @@ export function navigate(
     fetchPage = true,
     preventRemount,
     skipSetPath = false,
+    setPreview,
   } = options
 
   const navigationRoute = getNavigationRouteToNavigate(pages, options, true)
+
   if (navigationRoute.hash) {
     window.location.hash = navigationRoute.hash
+
     return true
   }
 
   if (navigationRoute == null) {
     return false
   }
 
+  if (fallbackToWindowLocation) {
+    setPreview && setPreview(true)
```

Comment:
How about

```suggestion
   setPreview?.(true)
```

Reply:
I liked the explicit way that kaisermann recommended above, what do you think?

Reply:
That's fine! 😄